packages/p2p/src/components/buy-sell/buy-sell-modal.jsx

```diff
@@ -143,6 +168,26 @@ const BuySellModal = ({ table_type, selected_ad, should_show_popup, setShouldSho
 
     const setSubmitForm = submitFormFn => (submitForm.current = submitFormFn);
 
+    React.useEffect(() => {
+        const disposer = reaction(
```

Comment:
Used `reaction(...)` instead of `when(buy_sell_store.form_error_code === api_error_codes.MARKET_RATE_CHANGE)` since in a scenario where the rate changes twice, the error code `MARKET_RATE_CHANGE` would be the same, thus the `when` reaction would not be executed by comparing previous `buy_sell_store.form_error_code` and current new value and deeming it to be not changed